scripts/parse-coverage.sh

```
@@ -0,0 +1,175 @@
+
+# Package-level function coverage map
+declare -A pkg_func_pct
+while IFS: read -r line; do
+  [[ "$line" =~ ^total: ]] && continue
+  [[ -z "$line" ]] && continue
+
+  # Parse: "github.com/poyrazk/thecloud/path/file.go:line.col  FuncName  coverage%"
+  file_path=$(echo "$line" | awk '{print $1}')
+  coverage=$(echo "$line" | awk '{print $3}' | tr -d '%')
+
+  pkg="${file_path#$MODULE_PREFIX}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)